### PR TITLE
Add ifndef clause

### LINKED ISSUE #1278: Use a wrapper #ifndef in cryptographicsHash.c
This is needed for Spawn of EnergyPlus to avoid that kind of compilation error:
```
cryptographicsHash.c:55:6: error: redefinition of ‘SHA1Transform’
```

### PR DESCRIPTION
**Pull request by AntoineGautier**
This closes #1278.

### CODE REVIEW

File: IBPSA/Resources/C-Sources/cryptographicsHash.c [hunk 1]
```diff
@@ -312,3 +313,5 @@ const char* cryptographicsHash(const char* str)
 
   return hexresult;
 }
+
+#endif
```
Comment (Mathadon) [inline IBPSA/Resources/C-Sources/cryptographicsHash.c]: @AntoineGautier looks like this needs a newline?

Comment (AntoineGautier) [reply IBPSA/Resources/C-Sources/cryptographicsHash.c]: @Mathadon I did not know that. Is is related to [https://stackoverflow.com/questions/72271/no-newline-at-end-of-file-compiler-warning](https://stackoverflow.com/questions/72271/no-newline-at-end-of-file-compiler-warning)?
(I added it to fileWriterStructure.c as well.)

Comment (Mathadon) [reply IBPSA/Resources/C-Sources/cryptographicsHash.c]: yup, it's a bit of a detail but a nice convention to uphold when coding

**Comment by AntoineGautier** [comment]
So this should now be ready to be merged.

**Pull request merged**